src/main/groovy/lang/MetaClassImpl.java

```diff
@@ -1832,6 +1832,9 @@ public Object getProperty(Class sender, Object object, String name, boolean useS
             } catch (IllegalArgumentException e) {
                 // can't access the field directly but there may be a getter
                 mp = null;
+            } catch (GroovyRuntimeException e) {
```

Comment:
I think AccessPermissionChecker should only throw exceptions extending from AccessControlException. because it specifically checks access permissions. 

So I suggest to introduce `public class CacheAccessControlException extends AccessControlException`